Source/GUI/MemViewer/MemViewer.cpp

```diff
@@ -131,186 +121,275 @@ void MemViewer::changeMemoryRegion(const bool isMEM2)
 {
   m_memViewStart = isMEM2 ? Common::MEM2_START : Common::MEM1_START;
   m_memViewEnd = isMEM2 ? Common::MEM2_END : Common::MEM1_END;
-  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / 16) - 1);
+  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / VISIBLE_COLS) - VISIBLE_ROWS);
 }
 
 void MemViewer::mousePressEvent(QMouseEvent* event)
 {
+  // Only handle left-click events
+  if (event->button() != Qt::MouseButton::LeftButton)
+    return;
+
   int x = event->pos().x();
   int y = event->pos().y();
-  if (x >= m_rowHeaderWidth && x <= m_rowHeaderWidth + m_hexAreaWidth && y >= m_columnHeaderHeight)
+
+  const bool wasEditingHex = m_editingHex;
+  const int spacing = m_charWidthEm / 2;
+  int clickedHexPosX = x - m_rowHeaderWidth + spacing / 2;
+  int clickedAsciiPosX = x - (m_hexAsciiSeparatorPosX + spacing);
+  int clickedPosY = y - m_columnHeaderHeight;
+
+  // Ignore clicking the headers
+  if (clickedPosY < 0 || clickedHexPosX < 0)
+    return;
+
+  // Ignore space between HEX and ASCII table
+  if (clickedHexPosX > m_hexAreaWidth && clickedAsciiPosX < 0)
+    return;
+
+  // Transform click positions to table positions
+  clickedHexPosX /= spacing + m_charWidthEm * 2;
+  clickedAsciiPosX /= m_charWidthEm;
+  clickedPosY /= m_charHeight;
+
+  // Ignore clicking the padding at the bottom
+  if (clickedPosY >= VISIBLE_ROWS)
+    return;
```

Comment:
I don't like this way of doing this for 2 reasons: 1: it doesn't consider the horizontal space after the ascii area and second, it's honestly more confusing to read to make sure this work because essentially, what you want to do is test that the click was inside the grid and having all these conditions separated doesn't make that part obvious.  I would rather keep the 4 conditions because they all set each boundaries of the grid.